### PR TITLE
Add array json

### PR DESCRIPTION
**Pull request by jiaoew1991**

### CODE REVIEW

**Comment by jiaoew1991** [comment]
issue: https://github.com/milvus-io/milvus/issues/23389

**Comment by jiaoew1991** [comment]
issue: https://github.com/milvus-io/milvus-proto/issues/128

File: proto/schema.proto [hunk 1]
```diff
@@ -103,6 +97,8 @@ message ScalarField {
     DoubleArray double_data = 5;
     StringArray string_data = 6;
     BytesArray bytes_data = 7;
+    ArrayArray array_data = 8;
```
Comment (xiaofan-luan) [inline proto/schema.proto:100]: Let's not work the Array of Array and JSON Array since that would be too much?

Comment (jiaoew1991) [reply proto/schema.proto:100]: The arrays of other databases only support a single type, and a single-type array can be optimized for more operations. In addition, JSON arrays can be represented using the JSON data type.

**Pull request merged**